### PR TITLE
CB-8987 Extend Distrox and Sdx API endpoints with rotateCertificates method

### PR DESCRIPTION
**Pull request by lacikaaa**
This commit adds the rotateCertificates to the distrox and sdx endpoints. Also creates a new flow
in datalake service which initiates the flow in CB and wait until it finishes.
`datahub/rotateCertDatahub` and `datalake/rotateCertDatalake` are introduced new authorization
resource actions.

### CODE REVIEW

File: authorization-common-api/src/main/java/com/sequenceiq/authorization/resource/AuthorizationResourceAction.java [hunk 1]
```diff
@@ -45,6 +45,7 @@
     START_DATALAKE("datalake/startDatalake", AuthorizationResourceType.DATALAKE),
     STOP_DATALAKE("datalake/stopDatalake", AuthorizationResourceType.DATALAKE),
     UPGRADE_DATALAKE("datalake/upgradeDatalake", AuthorizationResourceType.DATALAKE),
+    ROTATE_CERT_DATALAKE("datalake/rotateCertDatalake", AuthorizationResourceType.DATALAKE),
```
Comment (keyki) [inline authorization-common-api/src/main/java/com/sequenceiq/authorization/resource/AuthorizationResourceAction.java:48]: should we call it rotateTlsCertDatalake instead?

**Review by horadla23 — CHANGES_REQUESTED** [review]
since it is new API and new right, it is OK, but it would better if you could add new rights to UMS first, then merge this

File: authorization-common-api/src/main/java/com/sequenceiq/authorization/resource/AuthorizationResourceAction.java (hunk 1 above)
Comment (lacikaaa) [reply authorization-common-api/src/main/java/com/sequenceiq/authorization/resource/AuthorizationResourceAction.java]: renamed

Comment (sodre90) [reply authorization-common-api/src/main/java/com/sequenceiq/authorization/resource/AuthorizationResourceAction.java]: please update legacyRights.json also

Comment (lacikaaa) [reply authorization-common-api/src/main/java/com/sequenceiq/authorization/resource/AuthorizationResourceAction.java]: right, I will upload with the unit tests

Comment (lacikaaa) [reply authorization-common-api/src/main/java/com/sequenceiq/authorization/resource/AuthorizationResourceAction.java]: updated and unit test added

File: datalake/src/main/java/com/sequenceiq/datalake/service/sdx/cert/CertRotationService.java [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+import com.sequenceiq.datalake.service.sdx.CloudbreakFlowService;
+import com.sequenceiq.datalake.service.sdx.CloudbreakFlowService.FlowState;
+import com.sequenceiq.datalake.service.sdx.PollingConfig;
+import com.sequenceiq.datalake.service.sdx.SdxService;
+import com.sequenceiq.datalake.service.sdx.status.AvailabilityChecker;
+import com.sequenceiq.datalake.service.sdx.status.SdxStatusService;
+import com.sequenceiq.flow.api.model.FlowIdentifier;
+
+@Service
+public class CertRotationService {
```
[the remark this answers is not in the document]
Comment (lacikaaa) [reply datalake/src/main/java/com/sequenceiq/datalake/service/sdx/cert/CertRotationService.java:30]: I have added some unit tests

File: core-api/src/main/java/com/sequenceiq/distrox/api/v1/distrox/endpoint/DistroXV1Endpoint.java [hunk 3]
```diff
@@ -359,4 +362,16 @@ void setClusterMaintenanceModeByCrn(@PathParam("crn") String crn,
     @ApiOperation(value = RENEW_CERTIFICATE, produces = MediaType.APPLICATION_JSON, notes = Notes.RENEW_CERTIFICATE_NOTES,
             nickname = "renewDistroXCertificate")
     FlowIdentifier renewCertificate(@PathParam("crn") String crn);
+
+    @PUT
+    @Path("name/{name}/rotate_certificates")
```
Comment (keyki) [inline core-api/src/main/java/com/sequenceiq/distrox/api/v1/distrox/endpoint/DistroXV1Endpoint.java:367]: Should we call it rotate_autotls_certificates similarly to the permissions?

Comment (lacikaaa) [reply core-api/src/main/java/com/sequenceiq/distrox/api/v1/distrox/endpoint/DistroXV1Endpoint.java:367]: it's coming from `StackV4Endpoint`. I can change this, but then I think I would change in `StackV4Endpoint` too to keep them in sync. Shall I change them?

Comment (keyki) [reply core-api/src/main/java/com/sequenceiq/distrox/api/v1/distrox/endpoint/DistroXV1Endpoint.java:367]: yes, please.

**Pull request merged**